plonetheme/barceloneta/theme/rules.xml

```diff
@@ -33,8 +33,10 @@
   <!-- move global nav -->
   <replace css:theme-children="#mainnavigation" css:content-children="#portal-mainnavigation" method="raw" />
 
-  <!-- include view @@hero on homepage only -->
-  <after css:theme="#mainnavigation-wrapper" css:content=".principal" href="/@@hero" css:if-content="body.template-document_view.section-front-page" />
+  <!-- move hero to top -->
+  <replace css:theme-children="#hero .gigantic" css:content-children=".hero"/>
+  <drop css:theme="#hero" css:if-not-content=".hero" />
+  <drop css:content=".hero" />
```

Comment:
Sure, this works? Regarding http://docs.diazo.org/en/latest/basic.html#order-of-rule-execution ``<drop>`` rules are executed before ``<replace>`` rules. Looks to me, wether you'd have to remove the ``<drop>`` rule or add a ``method="raw"`` to your ``<replace>`` rule.
In either case you'd need to remove the ``method="raw"`` attribute of the replace-rule which copies the main content (also discussed here: https://community.plone.org/t/adding-html-snippets-with-diazo/1554/15 and https://github.com/plone/plonetheme.barceloneta/issues/85 )

Sorry, didn't try out your pull, so this might be wrong...